### ISSUE TITLE
How to get  88.40% accuracy with ConvNet network?

### ISSUE THREAD

**Issue by wonwooddo** [mmlind/mnist-dnn#2]
Hi, I have a question but, dont know how to contact you,  so posted the issue here.

Firstly, I really appreciate your work as I'm studying neural network, and try to implement in C.
it's really helpful to study by your blog and code, thankyou.

My question is  how did you construct your network to get high performance by using convolution network, like you showed on the blog (like below).

==========================================================
| Network     | # of Nodes     | Act.Fct | L-Rate | # Trains | Speed* | Accuracy || Compare** |
| ----------- |----------------|---------|--------|----------|--------|----------||-----------|
| 1-l   FC NN |      10 output | SIGMOID | 0.0125 |  900,000 |   228s |   91.09% ||    88.00% |     
| 2-l   FC NN |     300 hidden | SIGMOID | 0.0700 |  120,000 |   697s |   94.67% ||    96.40% |     
| 3-l   FC NN | 500+150 hidden | SIGMOID | 0.0050 |  180,000 | 2,510s |   93.64% ||    97.05% |     
**| 3-l ConvNet | 13x13x5, 6x6x5 |  RELU   | 0.0004 |  525,000 |   908s |   88.40% ||    98.90% |**     

==========================================================

When I tried the same 13x13x5, 6x6x5 convolutional network, but keep getting 9.87%.

I cloned your code and simply run as your default main.c has 13x13x5, 6x6x5 convolutional network.

and one more thing,
when try to use relu as activation function, it also do not work, keep getting around 9% accuracy
for just simply 1-hidden layer with relu neurons, I'm keep  varifing you code but dont get why.

Can you please help me with usage of your convolutional network code?

**Comment by liangam** [comment]
Hi  wonwooddo,

For improving the performance, the simple attempt you could do is to change the learning rate 
nn->learningRate = 0.07;

MNIST-DNN: A deep neural network for MNIST image recognition 

---------------------------------------------------------------------------------------------------------------------
Layer Index                    |        0       |        1       |        2       |        3       ||     TOTAL     |
Layer Type                     |      INPUT     |  CONVOLUTIONAL |  CONVOLUTIONAL |     OUTPUT     ||               |
---------------------------------------------------------------------------------------------------------------------
Activation Function            |                |     SIGMOID    |     SIGMOID    |     SIGMOID    ||               |
Image Matrix (width x height)  |     28 x 28    |     13 x 13    |      6 x  6    |     10 x  1    ||               |
Feature Maps (depth)           |        1       |        5       |        5       |        1       ||               |
Filter Size                    |                |      5 x 5     |      3 x 3     |                ||               |
Stride                         |                |        2       |        2       |                ||               |
---------------------------------------------------------------------------------------------------------------------
Number of Nodes                |          784   |          845   |          180   |           10   ||        1819   |
Number of Connections          |            0   |        21125   |         8100   |         1800   ||       31025   |
Number of Weights              |            0   |          125   |          225   |         1800   ||        2150   |
Memory Size (bytes)            |        43944   |       982944   |       166216   |        29400   ||     1239752   |
---------------------------------------------------------------------------------------------------------------------

Initializing network... 

Training: Reading image No.  60000 of  60000 images [100%]  Result: Correct= 47417  Incorrect= 12583  Accuracy=79.03%
Testing:  Reading image No.  10000 of  10000 images [100%]  Result: Correct=  9051  Incorrect=   949  Accuracy=90.51%%

DONE! Total execution time: 35.0 sec

Thanks for Matt's great work to help me understand the DNN a lot !
